Modules/pyexpat.c

```diff
@@ -1881,9 +1959,19 @@ MODULE_INITFUNC(void)
 
     /* export using capsule */
     capi_object = PyCapsule_New(&capi, PyExpat_CAPSULE_NAME, NULL);
-    if (capi_object)
-        PyModule_AddObject(m, "expat_CAPI", capi_object);
-    return m;
+    if (capi_object) {
+        if (PyModule_AddObject(mod, "expat_CAPI", capi_object) < 0) {
+            goto error;
+        }
+    }
```

Comment:
You still leak codes_dict and rev_codes_dict here. Maybe call Py_CLEAR() just after each PyModule_AddObject() call using them?

To ease error handling, you might add a new subfunction which only adds "codes" and "messages" attributes to the module.

Reply:
@vstinner Yes this is true, if the function succeeds I never cleared it.  I fixed it by clearing after I'm done using them.

Reply:
@vstinner actually PyModule_AddObject decrefs these two dicts on success, so there is no leak.  When I removed it the tests failed from a SIGSEV